python-sdk/tests/sql/test_table.py

```diff
@@ -145,3 +146,8 @@ def test_temp_table(table):
     assert table.temp
     assert isinstance(table, TempTable)
     assert not isinstance(table, Table)
+
+
+def test_if_table_object_can_be_pickled():
+    """Verify if we can pickle Table object"""
+    pickle.loads(pickle.dumps(Table(name="test")))
```

Comment:
Let's add this test for File object. The reason `self.__dict__` is not available for `File` is because it uses `slots`.

>A class whose instances have no [object.__dict__](https://docs.python.org/3/library/stdtypes.html#object.__dict__) attribute and [define](https://docs.python.org/3/reference/datamodel.html#slots) their attributes in a [object.__slots__](https://docs.python.org/3/reference/datamodel.html#object.__slots__) attribute instead. In attrs, they are created by passing slots=True to @attr.s (and are on by default in [attrs.define()](https://www.attrs.org/en/stable/api.html#attrs.define)/[attrs.mutable()](https://www.attrs.org/en/stable/api.html#attr.attrs.mutable)/[attrs.frozen()](https://www.attrs.org/en/stable/api.html#attr.attrs.frozen)).

Reply:
Got it

Reply:
But atleast this test will verify that you can pickle and unpickle the object.

Reply:
Would the following be a better test?

```python
table = Table(name="test")
assert pickle.loads(pickle.dumps(table)) == table
```

cc @ashb

Reply:
@kaxil Ya this seems better.